### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by frankban**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+# concurrency-limiter
+
+Limit the number of asynchronous concurrent tasks running.
+
+## Goals
+
+Assume you need to connect to many different WebSocket servers, or retrieve
+many resources concurrently at the same time. Any asynchronous operation can be
+started easily with JavaScript, but when too many of them could potentially be
+executed at the same time it's usually a good idea to limit their number in
```
Comment (rogpeppe) [inline README.md]: s/in order //
?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+order to avoid resource consumption on the browser or the node instance.
+
+Limiting the number of concurrent asynchronous tasks running is exactly the
```
Comment (rogpeppe) [inline README.md]: s/exactly the //
?

**Review by rogpeppe — APPROVED** [review]
LGTM with a couple of trivial text simplification suggestions. Looks nice, thanks.

File: README.md (hunk 1 above)
Comment (frankban) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (frankban) [reply README.md]: Done

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+.PHONY: lint
+lint: setup
+	npm run lint
+
+.PHONY: release
+release: clean check
+	$(eval current := $(shell npm view $(PROJECT) version))
+	$(eval package := $(shell npm version | grep $(PROJECT) | cut -d "'" -f 2))
+	@test $(current) != $(package) || ( \
```
Comment (hatched) [inline Makefile:35]: This can be removed, npm won't allow you to publish the same version again.

**Review by hatched — APPROVED** [review]
This looks great @frankban.

Maybe in a follow-up we could add an example folder, I quite like that from our other projects.

File: Makefile (hunk 3 above)
Comment (frankban) [reply Makefile:35]: But it is still useful here because it protects tagging the release.

**Pull request merged**